pkg/cvo/metrics.go

```diff
@@ -194,9 +195,10 @@ func (m *operatorMetrics) Collect(ch chan<- prometheus.Metric) {
 					g.Set(float64(failing.LastTransitionTime.Unix()))
 				}
 				ch <- g
+				setFailing = true
 			}
 		}
-		if failing != nil && failing.Status == configv1.ConditionTrue {
+		if !setFailing && failing != nil && failing.Status == configv1.ConditionTrue {
```

Comment:
should we DRY it up with something like:
```go
if failing != nil && failing.Status == configv1.ConditionTrue {
    g := m.version.WithLabelValues("failure", current.Version, current.Image, completed.Version)
    if update := cv.Spec.DesiredUpdate; update != nil && update.Image != current.Image {
       g = m.version.WithLabelValues("failure",update.Version, update.Image, completed.Version)
    }
    if failing.LastTransitionTime.IsZero() {
        g.Set(0)
    } else {
        g.Set(float64(failing.LastTransitionTime.Unix()))
    }
    ch <- g
}
```

Reply:
Yeah that might be better